### PR TITLE
Change coverage on zero results to be 100%

### LINKED ISSUE #381: Error when --since does not select any subjects
When using `--since` with something like `HEAD~1` no subjects are selected and an error results. I think this shouldn't result in an error since you don't always know what the diff is going to contain, and it may occasionally be empty which shouldn't cause the build to break.

### PR DESCRIPTION
**Pull request by mbj**
[fix #381]

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb This was validated via:

```
bundle exec mutant --zombie -I lib -r mutant --use rspec --since 'HEAD^1' -- 'Mutant*'
```

And is ready for review.

File: lib/mutant/result.rb [hunk 1]
```diff
@@ -11,24 +11,11 @@ module Coverage
       #
       # @api private
       def coverage
-        return Rational(0) if amount_mutation_results.zero?
+        return Rational(1) if amount_mutation_results.zero?
```
Comment (dkubb) [inline lib/mutant/result.rb]: Should `Rational(1)` be a constant?

Comment (mbj) [inline lib/mutant/result.rb]: True. Will change that.

Comment (dkubb) [inline lib/mutant/result.rb]: Given that there is very little code in this method I wonder if it would be cleaner to write it without a guard clause?

Comment (mbj) [inline lib/mutant/result.rb]: k

**Comment by mbj** [comment]
@dkubb Under the assumption you finished your last pass, its ready for the next pass, all flags addressed.

**Comment by dkubb** [comment]
@mbj hmm, for some reason this isn't working for me in our work project.

```
RAILS_ENV=test bundle exec mutant -r./config/environment --use rspec --since HEAD~1 -- ApplicationController

...

Mutant configuration:
Matcher:         #<Mutant::Matcher::Config match_expressions: [ApplicationController] subject_filters: [#<Mutant::Repository::SubjectFilter diff=#<Mutant::Repository::Diff from="HEAD" to="HEAD~1">>]>
Integration:     Mutant::Integration::Rspec
Expect Coverage: 100.00%
Jobs:            8
Includes:        []
Requires:        ["./config/environment"]
Subjects:        0
Mutations:       0
Kills:           0
Alive:           0
Runtime:         0.06s
Killtime:        0.00s
Overhead:        Inf%
Coverage:        0.00%
Expected:        100.00%
```

The status code was 0. And just to confirm I was using this branch:

```
$ bundle show mutant
/Users/dkubb/.gem/ruby/2.1.6/bundler/gems/mutant-9088a4b778e2
```

**Comment by mbj** [comment]
@dkubb I checked our work project, and found *ugly* bundler behavior:

Our Gemfile in the branch includes:

```
gem 'mutant',       '~> 0.8.0', git: 'https://github.com/mbj/mutant.git'
gem 'mutant-rspec', '~> 0.8.0', git: 'https://github.com/mbj/mutant.git', branch: 'fix/exit-on-zero-subjects'
```

Some requires will happen against `master`, and some against the branch: When *both* gems reference the `fix/exit-on-zero-subjects` the behavior is absent.

**Comment by tjchambers** [comment]
@mbj As an aside I have been a bit confused why mutant-rspec and mutant are in the same github project. It seems with the intro of minitest support they should be separated, but I am a noob at this and don't know what the ramifications of that are. I am sure that there is a good reason (you folks always have GREAT reasons), but thought I would comment.

**Comment by mbj** [comment]
> @mbj As an aside I have been a bit confused why mutant-rspec and mutant are in the same github project.

Because each repo is a maintenance overhead. And the implementation of `mutant-rspec` is to small to mandate the overhead of another repository.

**Comment by tjchambers** [comment]
I expected a good reason and got one. Thanks.

**Comment by mbj** [comment]
@dkubb this is ready for the next pass.

**Pull request merged**